File: 模式pk/门面模式VS中介者模式/门面模式-实现HR系统/Context.java
//门面不参与子系统内的业务逻辑
public class Context{
    //委托处理
    private ClassA a = new ClassA();
    private ClassB b = new ClassB();
    //复杂的计算
    public void complexMethod(){
        this.a.doSomethingA();
        this.b.doSomethingB();
    }
}

File: 通用代码/门面模式(C++)/Context.cpp
#include "Context.h"

Context::Context()
{
    this->_a = new ClassA();
    this->_b = new ClassB();
    cout << "Context..." << endl;
}

Context::~Context()
{
    delete this->_a;
    delete this->_b;

    this->_a = NULL;
    this->_b = NULL;
    cout << "~Context..." << endl;
}

void Context::complexMethod()
{
    cout << "委托处理复杂的计算" << endl;
    this->_a->doSomethingA();
    this->_b->doSomethingB();
}
